inc/blocks/broadcast/edit.js

```diff
@@ -0,0 +1,228 @@
+import debouncePromise from 'debounce-promise';
+import _deburr from 'lodash/deburr';
+import { v4 as uuid } from 'uuid';
+
+import { Fragment, useState, useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+
+import BroadcastIcon from './icon.js';
+
+const {
+	Button,
+	Icon,
+	TextControl,
+	Spinner,
+} = wp.components;
+
+const {
+	InspectorControls,
+} = wp.blockEditor;
+
+const apiFetch = wp.apiFetch;
+
+/**
+ * Block edit component.
+ *
+ * @param {*} props
+ *
+ * @returns {ReactElement}
+ */
+const EditComponent = function ( props ) {
+	const {
+		attributes,
+		setAttributes,
+	} = props;
+
+	const [ broadcast, _setBroadcast ] = useState( null );
+	const [ broadcasts, setBroadcasts ] = useState( [] );
+	const [ isFetching, setIsFetching ] = useState( false );
+	const [ searchKeyword, _setSearchKeyword ] = useState( '' );
+
+	const fetchBroadcasts = debouncePromise( function ( keyword, extraArgs ) {
+		const args = {
+			path: addQueryArgs( '/accelerate/v1/broadcast', { search: keyword } ),
+			parse: false,
+			...extraArgs,
+		};
+
+		return apiFetch( args )
+			.then( response => response.json ? response.json() : [] );
+	}, 1000 );
+
+	const fetchBroadcast = debouncePromise( function ( id, extraArgs ) {
+		const args = {
+			path: addQueryArgs( `accelerate/v1/broadcast/${ id }` ),
+			parse: false,
+			...extraArgs,
+		};
+
+		return apiFetch( args ).then( response => response.json ? response.json() : null );
+	}, 1000 );
+
+	/**
+	 * Select a Broadcast block.
+	 *
+	 * @param {*} broadcast Broadcast block metadata
+	 * @returns {void}
+	 */
+	const setBroadcast = function ( broadcast ) {
+		if ( ! broadcast && ! window.confirm( __( 'Are you sure you want to replace this Broadcast?', 'altis' ) ) ) {
+			return;
+		}
+		_setBroadcast( broadcast );
+		setAttributes( { broadcast: broadcast?.id } );
+	};
+
+	/**
+	 * Normalize and set search keyword.
+	 *
+	 * @param {string} keyword Search keyword.
+	 */
+	const setSearchKeyword = function ( keyword ) {
+		// Disregard diacritics.
+		//  Input: "média"
+		keyword = _deburr( keyword );
+
+		// Accommodate leading slash, matching autocomplete expectations.
+		//  Input: "/media"
+		keyword = keyword.replace( /^\//, '' );
+
+		// Strip leading and trailing whitespace.
+		//  Input: " media "
+		keyword = keyword.trim();
+
+		_setSearchKeyword( keyword );
+	};
+
+	// Set the block clientId if none currently set.
+	useEffect( () => {
+		if ( ! attributes.clientId ) {
+			setAttributes( { clientId: uuid() } );
```

Comment:
Could look at the experience block example- blocks already have a UUID a level above the attributes so you can read from there instead of generating one. Not a blocker, just an alternative approach that means you don’t need the uuid library.

Also I learned recently browsers support this natively now: `window.crypto.randomUUID();`

https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID

Reply:
Makes sense, each block does have a client-id already. Will remove the attribute then.

Reply:
Hmmm, seems like the clientId isn't persistent between sessions, so will keep the attribute but use the client id to fill it in instead of generating one.